### ISSUE TITLE
Support for Debian "trixie" release in docker-outside-of-docker feature

### ISSUE THREAD

**Issue by dalybrown** [devcontainers/features#1059]
Hello,

Is there a reason why the testing release of Debian (`trixie` or `sid`) is not supported in this list?

https://github.com/devcontainers/features/blob/648c8d56b9ec4f90cb9d754cd532a9d2e08c0674/src/docker-outside-of-docker/install.sh#L23

I couldn't find any information on why the codename needs to be checked. For development, we like to use the `sid` repository for many packages, but that will update our `/etc/os-release` information to `trixie` (currently) and therefore won't be supported. Consequently, the feature can't be installed.

Can we add `trixie` to this list? Thanks!

**Comment by erwinkramer** [comment]
Also wondering this for the `docker-in-docker` feature, let me know if i need to make a separate issue for that. Want to run this inside image `mcr.microsoft.com/devcontainers/dotnet:dev-10.0-preview-trixie`

**Comment by skatsubo** [comment]
As Trixie went stable, it's time to update the abovementioned `DOCKER_LICENSED_ARCHIVE_VERSION_CODENAMES`, right?

@dalybrown 
> I couldn't find any information on why the codename needs to be checked.

I guess it's for installing from download.docker.com
https://github.com/devcontainers/features/blob/849a5e2a7fd2c8109531cec9e63bfde12472407a/src/docker-outside-of-docker/install.sh#L221-L236

Actually Trixie on download.docker.com became available since April https://github.com/moby/moby/issues/49494

**Comment by erwinkramer** [comment]
@skatsubo, @AlvaroRausell it's kind of supported now, but not with moby enabled:

```
10.72 E: Unable to locate package moby-cli
10.72 E: Unable to locate package moby-buildx
10.72 E: Unable to locate package moby-engine
10.73 (!) Packages for moby not available in OS debian trixie (amd64).
```

**Comment by skatsubo** [comment]
> but not with moby enabled

Right.
`moby: true` is not working currently for trixie
> due to missing moby package in packages.microsoft.com/repos/microsoft-debian-trixie-prod

Discussed in https://github.com/devcontainers/features/pull/1442

**Comment by threema-donat** [comment]
@skatsubo Is it correct that this is still not working with trixie although the merge request #1442 is completed? I'm trying to build a dev container with trixie and it says:
```
E: Unable to locate package moby-buildx
(!) It seems packages for moby not available in OS debian trixie (amd64). To resolve, either: (1) set feature option '"moby": false' , or (2) choose a compatible OS version (eg: 'ubuntu-24.04').
[2025-09-01T05:38:56.758Z] ERROR: Feature "Docker (docker-outside-of-docker)" (ghcr.io/devcontainers/features/docker-outside-of-docker) failed to install! Look at the documentation at https://github.com/devcontainers/features/tree/main/src/docker-outside-of-docker for help troubleshooting this error.
```

**Comment by skatsubo** [comment]
@threema-donat 
> Is it correct that this is still not working with trixie although the merge request [#1442](https://github.com/devcontainers/features/pull/1442) is completed?

Right. You can follow suggestions in the error message for a workaround.
More details in the linked PR and in comments above.

**Comment by threema-donat** [comment]
Sorry for the dumb question, but I do not know where the `"moby": false` option can be set. I'm building a container myself from a Dockerfile and cannot find an option in the build section to place this key. Is it meant to be set as part of the "options" key?

Will this feature be released soon now that it is merged, so that I can just wait for it to arrive?
